### PR TITLE
The vampire's "Rejuvenate" ability will, if the vampire is sufficiently powerful, also stop any bleeding.

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
Initially I wanted to add this as one of the functions of a wooden coffin because it already fixes the vampire really good but I decided it's cooler and more dynamic this way.
With a better mastery over blood they can simply stitch their wounds shut. This does NOT heal the wounds themselves, but simply prevents any bleeding. A vampire won't let go of their own blood so easily!

Added a "post-upgrade" proc to vampire power datums, so that you can do more with them other than just giving descriptions and sometimes spells.

The "Rejuvenate" ability will now also have a modified description in the list of spells as well as provide an updated message if the vampire is powerful enough. It also gives you a count of how many wounds were stitched shut by the ability if you have any bleeding wounds, because why not?

:cl:
 * rscadd: The vampire's "Rejuvenate" ability will now, if the vampire is strong enough (at least 200 total blood), also close any bleeding wounds. This includes internal bleeding.
 * rscadd: The vampire's "Rejuvenate" ability now has a different description and activation message when it is empowered.

### CODE REVIEW

File: code/datums/gamemode/powers/vampire.dm [hunk 1]
```diff
@@ -52,9 +52,16 @@
...
 /datum/power/vampire/heal
 	cost = 200
-	granttext = "Your rejuvenation abilities have improved and will now heal you over time when used."
+	granttext = "Your rejuvenation abilities have improved and will now heal you over time when used, as well as stop all bleeding, including internal ones."
 	store_in_memory = TRUE
 
+/datum/power/vampire/heal/post_upgrade()
+	var/mob/M = role.antag.current
+	for(var/spell/rejuvenate/R in M.spell_list)
+		if(istype(R))
```
Comment (Exxion) [inline code/datums/gamemode/powers/vampire.dm]: It is not possible for this to be false. The loop already checks it.

**Comment by githubuser4141** [comment]
Why is blood loss lethal to a vampire? Surely it should only cause them to lose their performative blood, the blood that allows them to cast spells and use abilities, not actually affect their "homeostasis" because blood in a vampire doesn't actually circulate and so organs/tissues don't require any blood. Blood just pools in their chest cavity or something so they shouldn't be able to bleed through non-torso wounds either?

Sure, a vampire might die from blood loss but only when it's severely low (under 10%) and for an extended time period, it's more akin to dying from a lack of food (presumably there's something to remove blood from the vamp's tank and process it into fuel that doesn't involve circulation)? Internal bleeding makes no sense when all the blood's just pooled in the chest cavity.

**Comment by wotisjanitor** [comment]
if you touch vampires again, i recommend you add a limit of like 2 to thralls

**Comment by wotisjanitor** [comment]
githubuser, for all intents and purposes, vampires are just guys who have powers, weaknesses, and uncontrollable homicidal tendencies. In the Dracula novel by Bram Stoker, dracula is stabbed to death with knives. I know we don't go 1:1 with any source material but it's established that outside of their powers and weaknesses they have the same durability as regular people.

**Comment by SonixApache** [comment]
> Why is blood loss lethal to a vampire? Surely it should only cause them to lose their performative blood, the blood that allows them to cast spells and use abilities, not actually affect their "homeostasis" because blood in a vampire doesn't actually circulate and so organs/tissues don't require any blood. Blood just pools in their chest cavity or something so they shouldn't be able to bleed through non-torso wounds either?
> 
> Sure, a vampire might die from blood loss but only when it's severely low (under 10%) and for an extended time period, it's more akin to dying from a lack of food (presumably there's something to remove blood from the vamp's tank and process it into fuel that doesn't involve circulation)? Internal bleeding makes no sense when all the blood's just pooled in the chest cavity.

they have the same physiology as normal humans. So much like normal humans, bloodloss is the most common killer.

**Pull request merged**